<launch>

  <arg name="ground_truth_vertex_labels_file" 
    default="/media/giuliano/wd_ssd/datasets/replica/room_2/habitat/vertex_panoptic_labels.txt"/>
  <arg name="ground_truth_pointcloud_file" 
    default="/media/giuliano/wd_ssd/datasets/replica/room_2/habitat/mesh_semantic.ply"/>
  <arg name="experiments_dir"
    default="/media/giuliano/wd_ssd/planning_experiments/replica/room_2"/>
  <arg name="export_mesh" default="true"/>

  <node name="evaluation_manager_node" pkg="panoptic_mapping_evaluation" type="evaluation_manager_node.py" required="true">
    <param name="data_dir" value="$(arg experiments_dir)"/>
    <param name="export_eval_data_srv_name" value="export_evaluation_data"/>
    <param name="ground_truth_vertex_labels_file" value="$(arg ground_truth_vertex_labels_file)"/>
  </node>

  <node name="evaluation_data_export" pkg="panoptic_mapping_evaluation" type="evaluation_data_export_node">
    <param name="ground_truth_pointcloud_file" value="$(arg ground_truth_pointcloud_file)"/>
    <param name="is_single_tsdf" value="false"/>
    <param name="export_mesh" value="$(arg export_mesh)"/>
    <param name="refine_alignment" value="true"/>
    <rosparam param="alignment_transformation">
      [0.0, -1.0, 0.0, 0.0,
       1.0, 0.0, 0.0, 0.0,
       0.0, 0.0, 1.0, 0.0,
       0.0, 0.0, 0.0, 1.0]
    </rosparam>
  </node>
</launch>